src/nvim/terminal.c

```diff
@@ -278,8 +278,8 @@ void terminal_close(Terminal *term, char *msg, int status)
   buf_T *buf = handle_get_buffer(term->buf_handle);
   term->closed = true;
 
-  if (!msg || exiting) {
-    // If no msg was given, this was called by close_buffer(buffer.c).  Or if
+  if (status == -1 || exiting) {
+    // If status is -1, this was called by close_buffer(buffer.c).  Or if
```

Comment:
I think status can be -1 when it's coming from job control. So now we need a special boolean parameter, `close_buf` or something 😬

Reply:
Can it? The status code is an unsigned char so even if you try e.g. `exit -1` it still returns status code 255.

Reply:
-1 is used internally and by libuv, but I think it won't reach this codepath (which I guess is what you're saying).